pkg/diskmaker/diskmaker.go

```diff
@@ -184,24 +182,21 @@ func (d *DiskMaker) createSymLink(deviceNameLocation DiskLocation, symLinkPath s
 }
 
 func (d *DiskMaker) symLinkDisks(diskConfig *DiskConfig) {
-	// run command lsblk --all --noheadings --pairs --output "KNAME,PKNAME,TYPE,MOUNTPOINT"
-	// the reason we are using KNAME instead of NAME is because for lvm disks(and may be others)
-	// the NAME and device file in /dev directory do not match.
-	cmd := exec.Command("lsblk", "--all", "--noheadings", "--pairs", "--output", "KNAME,PKNAME,TYPE,MOUNTPOINT")
-	var out bytes.Buffer
-	var err error
-	cmd.Stdout = &out
-	err = cmd.Run()
+	blockDevices, badRows, err := internal.ListBlockDevices()
 	if err != nil {
-		msg := fmt.Sprintf("error running lsblk: %v", err)
+		msg := fmt.Sprintf("failed to list block devices: %v", err)
 		e := NewEvent(ErrorRunningBlockList, msg, "")
 		d.eventSync.Report(e, d.localVolume)
-		klog.Errorf(msg)
+		klog.Errorf(msg, "could not list block devices", "lsblk.BadRows", badRows)
 		return
+	} else if len(badRows) > 0 {
+		msg := fmt.Sprintf("error parsing rows: %+v", badRows)
+		e := NewEvent(ErrorRunningBlockList, msg, "")
+		d.eventSync.Report(e, d.localVolume)
+		klog.Errorf(msg, "could not parse all the lsblk rows", "lsblk.BadRows", badRows)
 	}
-	deviceSet := d.findNewDisks(out.String())
```

Comment:
Is `findNewDisks` not used anymore?  If that's the case, then this method implementation can be removed. Also, removing this method would make all the other methods in `disktable.go` redundant. So they should be removed as well. 

Also, the existing functionality was filtering the disks by ignoring disks based on whether they had `mountpoints` or if they were `root` disks with `partitions`.  See the `filterUsableDisks` method.  I don't see these filters applied after the new change. Is that expected ?

Reply:
Thanks Santosh for pointing it out, i agree that `disktable.go` will not be used anywhere when `findNewDisks` is removed and have modified accordingly. Also, made changes to preserve the existing functionality.